### PR TITLE
Feature/mirage

### PR DESCRIPTION
**Pull request by rahulkenchi**
create mock backend via mirage which will give us api response.

### CODE REVIEW

File: src/Miragejs/index.test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import makeServer from "./index";
+jest.mock("axios");
+
+export const BASE_URL = "http://localhost:3000/api";
```
Comment (jeffNeosoft) [inline src/Miragejs/index.test.js:4]: Need to keep this in the constants folder

Comment (rahulkenchi) [reply src/Miragejs/index.test.js:4]: will change this while adding test cases .

File: src/Miragejs/index.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            id: 7,
+          },
+        ],
+      });
+    },
+
+    routes() {
+      this.namespace = "/api";
+
+      this.get("/registeration", (schema, request) => {
```
Comment (santoshshinde2012) [inline src/Miragejs/index.js:83]: is this feasible to use `async await` ?

File: src/Miragejs/index.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        try {
+          let count;
+          console.log(request.queryParams);
```
Comment (santoshshinde2012) [inline src/Miragejs/index.js:86]: Remove `console.log`

File: src/Miragejs/index.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+            );
+          }
+          let tmp = schema.db.user.where(request.queryParams);
+          return new Response(200, { "Content-type": "application/json" }, tmp);
+        } catch (err) {
+          return new Response(500, { "Content-type": "application/json" });
+        }
+      });
+
+      this.put("/registeration/:id", (schema, request) => {
```
Comment (santoshshinde2012) [inline src/Miragejs/index.js:110]: optimize this function

**Review by santoshshinde2012 — APPROVED** [review]

File: src/Miragejs/index.js (hunk 3 above)
Comment (rahulkenchi) [reply src/Miragejs/index.js:86]: done.

File: src/Miragejs/index.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+            // console.log("tmp1", tmp);
+            if (request.queryParams.name_like != " ") {
+              let reg = new RegExp(request.queryParams.name_like);
+              tmp = tmp.filter((ele) => reg.test(ele.userid));
+            }
+            count = Math.ceil(tmp.length / 10);
+            tmp = tmp.splice((request.queryParams._page - 1) * 10, 10);
+            console.log("tmp3", tmp);
+            return new Response(
+              200,
```
[the remark this answers is not in the document]
Comment (rahulkenchi) [reply src/Miragejs/index.js:98]: Code optimized.

File: src/Miragejs/index.js (hunk 4 above)
Comment (rahulkenchi) [reply src/Miragejs/index.js:110]: Removed unnecessary code and console.logs.

File: src/app/config/Myservices.js [hunk 6]
```diff
@@ -2,23 +2,25 @@ import { URL } from "./URL";
 import axios from "axios";
 
 export function registerationGet(email, password) {
-  return axios.get(`${URL}/Registration?email=${email}&password=${password}`);
+  return axios.get(
+    `${URL}/api/registeration?email=${email}&password=${password}`
```
Comment (paras-1999) [inline src/app/config/Myservices.js:6]: change url with host

**Review by paras-1999 — APPROVED** [review]

File: src/Miragejs/index.js (hunk 2 above)
Comment (rahulkenchi) [reply src/Miragejs/index.js:83]: removed async await

File: src/app/config/Myservices.js (hunk 6 above)
Comment (rahulkenchi) [reply src/app/config/Myservices.js:6]: host variable is already used for another link.

File: src/index.js [hunk 7]
```diff
@@ -4,7 +4,12 @@ import { BrowserRouter } from "react-router-dom";
 import App from "./app/App";
 import { Provider } from "react-redux";
 import store from "./app/redux/store/store";
+import makeServer from "./Miragejs/index";
+
 // import "./i18n";
+
+makeServer();
```
Comment (paras-1999) [inline src/index.js:11]: can you add a comment about makeServer working

**Review by santoshshinde2012 — APPROVED** [review]

File: src/Miragejs/index.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+        }
+      });
+
+      this.put("/user/:id", (schema, request) => {
+        try {
+          const { params, requestBody } = request;
+          updateUser(schema, params, requestBody);
+          return responseUtils(200);
+        } catch (err) {
+          return responseUtils(500);
```
Comment (paras-1999) [inline src/Miragejs/index.js:117]: u passed 500 as a default parameter so no need to pass 500 over here

File: src/Miragejs/index.js [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+        }
+      });
+
+      this.post("/user", (schema, request) => {
+        try {
+          const { requestBody } = request;
+          userAdd(schema, requestBody);
+          return responseUtils(200);
+        } catch (err) {
+          return responseUtils(500);
```
Comment (paras-1999) [inline src/Miragejs/index.js:127]: same her n need to pass 500

**Review by paras-1999 — COMMENTED** [review]
fix the issues

**Pull request merged**